lecture-notes/3-bld.md

```diff
@@ -1085,8 +1111,39 @@ $ objdump -M intel --disassemble=main ./a.out
 $
 ```
 
-<!-- TODO(max): Also note what is going on for printf; DLL lookup through plt.
-What is DLL? What happens if we do -static? -->
+There are a couple things to note!
+
+The first one is that our code offsets got a lot bigger. Whereas before we
+started `main` with offset `0x0` and went until `0x24`, now we start at
+`0x1060` and go until `0x1084`. That seems to indicate that the `main` function
+has been bundled with other code, some of which has been stored before it.
+
+The second one is that the offset for calling `random_number` changed. The
+zero offset got filled in and is now `0d 01 00 00` (and therefore the text
+representation on the right changed too to point at `0x1179`, somewhere shortly
+after the end of `main`). Since x86 is "little endian", this number is
+`0x010d`, or 269 bytes after the next instruction (`0x106c+0x010d=0x1179`).
+
+We can check this by looking at the disassembly (and address) of the
+`random_number` function. Sure enough, it's at `0x1179`:
+
+```console?prompt=$
+$ objdump -M intel --disassemble=random_number ./a.out
+0000000000001179 <random_number>:
+    1179:	f3 0f 1e fa          	endbr64
+    117d:	b8 04 00 00 00       	mov    eax,0x4
+    1182:	c3                   	ret
+$
+```
+
+Last, we notice that the call to `printf` changed to this weird
+`__printf_chk@plt` thing. We're going to mostly gloss over that and say "it's
+dynamically linked", which means that at program start time, the loader will
+find and set the real address for `printf`.
+
+If you're curious, you can see what happens when we statically link our program
+with `-static`, which means that we want to bundle `printf` (and other normally
+dynamically linked libraries) alongside it:
```

Comment:
I think static linking should be a footnote. It's enough of a digression to not spend time on, IMO.

Reply:
footnoted